### PR TITLE
chore(all): fix DeepSource reported issues

### PR DESCRIPTION
**Pull request by qdm12**
## Changes

❗ Let's make DeepSource required after merging this

- [Hidden goroutine GO-E1007](https://deepsource.io/gh/ChainSafe/gossamer/run/6977cf21-542f-4d76-9244-a70b97854815/go/GO-E1007)
- [Using a deprecated function, variable, constant or field SCC-SA1019](https://deepsource.io/gh/ChainSafe/gossamer/run/6977cf21-542f-4d76-9244-a70b97854815/go/SCC-SA1019)
- [Found `x.Sub(time.Now())` instead of `time.Until(x)` SCC-S1024](https://deepsource.io/gh/ChainSafe/gossamer/run/6977cf21-542f-4d76-9244-a70b97854815/go/SCC-S1024)
- [Unused code SCC-U1000](https://deepsource.io/gh/ChainSafe/gossamer/run/6977cf21-542f-4d76-9244-a70b97854815/go/SCC-U1000)
- [Simplify slice expression to sliced value itself CRT-A0016](https://deepsource.io/gh/ChainSafe/gossamer/run/6977cf21-542f-4d76-9244-a70b97854815/go/CRT-A0016)
- [Control-coupled functions detected RVV-A0005](https://deepsource.io/gh/ChainSafe/gossamer/run/6977cf21-542f-4d76-9244-a70b97854815/go/RVV-A0005)
- [Exit inside non-main function detected RVV-A0003](https://deepsource.io/gh/ChainSafe/gossamer/run/6977cf21-542f-4d76-9244-a70b97854815/go/RVV-A0003)
- [Redefinition of built-in detected RVV-B0009](https://deepsource.io/gh/ChainSafe/gossamer/run/6977cf21-542f-4d76-9244-a70b97854815/go/RVV-B0009)
- [Unused method receiver detected RVV-B0013](https://deepsource.io/gh/ChainSafe/gossamer/run/6977cf21-542f-4d76-9244-a70b97854815/go/RVV-B0013?page=2)
- [`append` possibly assigns to a wrong variable CRT-D0001](https://deepsource.io/gh/ChainSafe/gossamer/run/6977cf21-542f-4d76-9244-a70b97854815/go/CRT-D0001)
- [Unused parameter detected in function RVV-B0012](https://deepsource.io/gh/ChainSafe/gossamer/run/6977cf21-542f-4d76-9244-a70b97854815/go/RVV-B0012)
- [Confusing naming of struct fields or methods RVV-B0001](https://deepsource.io/gh/ChainSafe/gossamer/run/6977cf21-542f-4d76-9244-a70b97854815/go/RVV-B0001)
- [Fix Importing the same package multiple times GO-W5021](https://deepsource.io/gh/ChainSafe/gossamer/run/a3a0c300-bb4b-484c-a732-1469eac07edb/go/GO-W5021)
- Rename trie node method `Type()` to `Kind()`

## Issues

#2347 

## Primary Reviewer

@edwardmack

### CODE REVIEW

**Review by EclesioMeloJunior — APPROVED** [review]

File: internal/trie/node/node.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ type Node struct {
 
 // Type returns Leaf or Branch depending on what type
 // the node is.
-func (n *Node) Type() Type {
+func (n *Node) Type() Kind {
```
Comment (jimjbrettj) [inline internal/trie/node/node.go:46]: I like this

**Review by jimjbrettj — APPROVED** [review]
Your friday nights are much more productive than mine

File: internal/trie/node/node.go (hunk 1 above)
Comment (qdm12) [reply internal/trie/node/node.go:46]: Oh wait, `Type()` should be `Kind()` now 😄 

It was just deepsource complaining that `type` is a built in name ugh

**Comment by qdm12** [comment]
> Your friday nights are much more productive than mine

😄 well after like 6 months wanting to do this, it felt as good as drinking beer

**Pull request merged**